File: src/test/java/dev/retrotv/crypt/owe/md/MD2Test.java
package dev.retrotv.crypt.owe.md;

import dev.retrotv.enums.Algorithm;
import dev.retrotv.crypt.owe.OWETest;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class MD2Test extends OWETest {

    @Test
    @DisplayName("MD2 File hash 테스트")
    void fileHashTest() throws Exception {
        fileHashTest(Algorithm.MD2);
    }

    @Test
    @DisplayName("MD2 File hash matches 테스트")
    void fileHashMatchesTest() throws Exception {
        fileHashMatchesTest(new MD2(), Algorithm.MD2);
    }

    @Test
    @DisplayName("MD2 File and File matches 테스트")
    void fileMatchesTest() throws Exception {
        fileMatchesTest(new MD2());
    }

    @Test
    @DisplayName("MD2 password encode 테스트")
    void passwordEncrypt() {
        passwordEncryptAndMatchesTest(new MD2());
    }
}

File: src/test/kotlin/dev/retrotv/crypto/owe/hash/md/MD2Test.kt
package dev.retrotv.crypto.owe.hash.md

import dev.retrotv.crypto.owe.OWETest
import dev.retrotv.enums.Algorithm
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

class MD2Test : OWETest() {
    @Test
    @DisplayName("MD2 File hash 테스트")
    fun fileHashTest() {
        fileHashTest(Algorithm.Hash.MD2)
    }

    @Test
    @DisplayName("MD2 File hash matches 테스트")
    fun fileHashMatchesTest() {
        fileHashMatchesTest(MD2(), Algorithm.Hash.MD2)
    }

    @Test
    @DisplayName("MD2 password encode 테스트")
    fun passwordEncrypt() {
        passwordEncryptAndMatchesTest(MD2())
    }
}
